aws/resource_aws_efs_file_system.go

```diff
@@ -160,6 +178,18 @@ func resourceAwsEfsFileSystemCreate(d *schema.ResourceData, meta interface{}) er
 	}
 	log.Printf("[DEBUG] EFS file system %q created.", d.Id())
 
+	_, hasLifecyclePolicy := d.GetOk("lifecycle_policy")
+	if hasLifecyclePolicy {
+		_, err := conn.PutLifecycleConfiguration(&efs.PutLifecycleConfigurationInput{
+			FileSystemId:      aws.String(d.Id()),
+			LifecyclePolicies: resourceAwsEfsFileSystemLifecyclePolicy(d),
```

Comment:
We typically prefer to reduce parameters to only the necessary data when creating functions like these to reduce complexity, e.g.

```suggestion
			LifecyclePolicies: resourceAwsEfsFileSystemLifecyclePolicy(d.Get("lifecycle_policy").([]interface{})),
```